src/methods.jl

```diff
@@ -0,0 +1,23 @@
+"""
+    bias_subtraction!(frame::AbstractArray, bias_frame::AbstractArray)
+
+In place version of [`bias_subtraction`](@ref)
+"""
+function bias_subtraction!(frame::AbstractArray, bias_frame::AbstractArray)
+    if size(frame) != size(bias_frame)
+        error("size of frame and bias_frame are not same")
+    end
+    for i in eachindex(frame)
+        @inbounds frame[i] = frame[i] - bias_frame[i]
+    end
+    return frame
```

Comment:
Did you benchmark broadcasting instead:
```suggestion
    return frame .-= bias_frame
```
?

Reply:
Haven't tried benchmarking broadcasting, I will give it a try! :smile:

Reply:
Broadcasting seems to be slower as compared to the for loop, the results are as follows:
```
function test1(frame1::AbstractArray, frame2::AbstractArray)
    return frame1 .-= frame2
end

function test2(frame1::AbstractArray, frame2::AbstractArray)
    for i in eachindex(frame1)
        @inbounds frame1[i] = frame1[i] - frame2[i]
    end
    return frame1
end

x = zeros(1000, 100)
y = rand(1000, 100)

@btime test1(y,x) # 44.443 μs (0 allocations: 0 bytes)
@btime test2(y,x) # 29.272 μs (0 allocations: 0 bytes)

```

Reply:
Uhm, I think that's because of `@inbounds`, if I remove that broadcasting is on-par, or even a bit faster.

Note that you [should interpolate variables](https://github.com/JuliaCI/BenchmarkTools.jl/blob/70d083edc497b12bcb405f1801cf03bffcf3840a/README.md#quick-start) when doing benchmarks with `BenchmarkTools.jl` macros.

Reply:
So, let's go with `@inbounds` implementation of the function. What would you suggest?

Reply:
It would be nice to use broadcasting, as it is GPU-friendly, but let's keep it as it is now for the time being.

Reply:
Sure, once the basic methods are written, we can then work upon the performance-based optimisations.